rsec16/coder.go

```diff
@@ -205,7 +211,8 @@ func (c Coder) ReconstructData(data, parity [][]byte) error {
 	}
 
 	if len(input) < c.dataShards {
-		return errors.New("not enough parity shards")
+		//not enough parity shards for repair
+		return IrrepairableError{}
```

Comment:
Also, this makes a test fail, right? Can you fix that test, and also add a similar check for `CanReconstructData`?